core/server/api/utils.js

```diff
@@ -118,7 +118,8 @@ utils = {
                 to: {isDate: true},
                 fields: {matches: /^[\w, ]+$/},
                 order: {matches: /^[a-z0-9_,\. ]+$/i},
-                name: {}
+                name: {},
+                email: {isEmail: true}
```

Comment:
That does not do anything, because `attrs` is getting transformed into `options.data` and the `options.data` object is validated separately, see comment below. There is a differentiation between data and options. Data is e.g. resources which you pass in the body AND dynamic route params e.g. /users/:email or /users/:slug.

Resource validations happens one level below (model layer, [see](https://github.com/TryGhost/Ghost/blob/1.17.1/core/server/models/base/index.js#L124) ), but for url data the validation has to happen [here](https://github.com/TryGhost/Ghost/blob/1.17.1/core/server/api/utils.js#L291);

Reply:
This is needed for the `apiUtils.validate` call otherwise the `email` param is treated as a `slug` which fails validation. See [comment here](https://github.com/TryGhost/Ghost/pull/9238#discussion_r150798659)

Reply:
Now it makes sense 👍